src/PAModel/Serializers/TransformLogo.cs

```diff
@@ -48,7 +48,7 @@ public static (PublishInfoJson, FileEntry) TransformLogoOnSave(this CanvasDocume
             FileEntry logoFile = null;
             var publishInfo = app._publishInfo.JsonClone();
 
-            if (!string.IsNullOrEmpty(publishInfo?.LogoFileName))
+            if (!string.IsNullOrEmpty(publishInfo?.LogoFileName) && app._logoFile?.Name != null)
```

Comment:
One potential gap I see in this new logic: what if `app._entropy.OldLogoFileName` has a value? Previously that would be used and preferred over the `app._logoFile.Name`, now it will be skipped. Is that desired?